Core/DISET/AuthManager.py

```diff
@@ -256,6 +262,25 @@ def getUsername(self, credDict):
       return True
     return False
 
+  def isUserSuspended(self, credDict):
+    """ Discover if the use is in Suspended status
+
+    :param dict credDict: Credentials to check
+    :return: Boolean True if user is Suspended
+    """
+    if self.KW_USERNAME not in credDict:
+      self.getUsername(credDict)
+    if self.KW_USERNAME not in credDict or self.KW_GROUP not in credDict:
+      return False
+    suspendedVOList = CS.getUserOption(credDict[self.KW_USERNAME], 'Suspended', [])
```

Comment:
This parts needs to be veeeeeery performant as Federico said. How about constructing a set (that you can refresh every so often) containing the list of suspended users per VO, instead of calling this every time ?

Reply:
Well, the new status can appear in the CS at any moment, then it should be something like DictCache used with a short expiration time and a whole logic to refresh it .But I think, this is not a lot of penalty to just make a string split where the option is already cached in the config client.

Reply:
I just wanted to make a performance test of this call, and realized that `getUserOption` does not belong to the CS module, that's strange no ?

Reply:
Just making a very quick test, the getUserOption is in the order of O(1000) slower than a set lookup. 
rather than a dictCache, I would just have a periodic task refreshing a set every now and then (you anyway only refresh the CS cache every so often).

Reply:
Well, I tried to make changes as you suggested with this piece of code:

```
  def isUserSuspended(self, credDict):
    """ Discover if the use is in Suspended status

    :param dict credDict: Credentials to check
    :return: Boolean True if user is Suspended
    """
    # Update credDict if the username is not there
    if self.KW_USERNAME not in credDict:
      self.getUsername(credDict)
    # If username or group is not known we can not judge if the user is suspended
    # These cases are treated elsewhere anyway
    if self.KW_USERNAME not in credDict or self.KW_GROUP not in credDict:
      return False
    self.__updateSuspendedUsers()
    vo = Registry.getVOForGroup(credDict[self.KW_GROUP])
    if "%s::%s" % (credDict[self.KW_USERNAME], vo) in self.suspendedUserSet:
      return True
    return False

  def __updateSuspendedUsers(self):

    if time.time() < self.lastSuspendedUpdate + 600:
      return
    self.suspendedUserSet = set()
    result = gConfig.getSections('/Registry/Users')
    userList = result['Value']
    for user in userList:
      voList = Registry.getUserOption(user, 'Suspended', [])
      for vo in voList:
        self.suspendedUserSet.add("%s::%s" % (user, vo))
    self.lastSuspendedUpdate = time.time()
```

I have not found any speed up, it is even a bit slower if even not to count the very first call to isSuspendedUser() where the set is built up. Typical call to isSuspendedUser() in both cases is about 0.2 msec, which I think is OK. May be you have a better idea to accelerate it, but I do not see how this can be done.